### PR TITLE
Add JEDI geoval variables for radiance data

### PR DESCRIPTION
**Pull request by emilyhcliu**
## Description

The following two types of variables are added:
(1) geoval variables required by CRTM
(2) Input observation error --- In the current UFO for clear-sky radiance assimilation, the input observation error (originally specified in GSI satinfo) is read from IODA file. This will be changed in the near future.

### CODE REVIEW

File: src/gsi/setuprad.f90 [hunk 1]
```diff
@@ -2573,13 +2576,14 @@ subroutine contents_netcdf_diag_(odiags,idv,iob)
...
-                 call nc_diag_metadata("Obs_Minus_Forecast_unadjusted",         sngl(tbcnob(ich_diag(i)))  )     ! observed - simulated Tb with no bias correction (K)
-                 call nc_diag_metadata("Obs_Minus_Forecast_adjusted",           sngl(tbc0(ich_diag(i)  ))  )     ! observed - simulated Tb with bias corrrection (K)
+                 call nc_diag_metadata("Observation",                           sngl(tb_obs0(ich_diag(i))) )    ! observed brightness temperature (K)
+                 call nc_diag_metadata("Obs_Minus_Forecast_unadjusted",         sngl(tbcnob(ich_diag(i)))  )    ! observed - simulated Tb with no bias correction (K)
+                 call nc_diag_metadata("Obs_Minus_Forecast_adjusted",           sngl(tbc0(ich_diag(i)  ))  )    ! observed - simulated Tb with bias corrrection (K)
                  errinv = sqrt(varinv0(ich_diag(i)))
                  call nc_diag_metadata("Inverse_Observation_Error",             sngl(errinv)          )
+                 call nc_diag_metadata("Input_Observation_Error",          sngl(error0(ich_diag(i)))  )         ! observed brightness temperature (K)
```
Comment (fcvdb) [inline src/gsi/setuprad.f90:2584]: ! brightness temperature observation error (K) ?

**Review by fcvdb — APPROVED** [review]
Thanks Emily!

File: src/gsi/setuprad.f90 (hunk 1 above)
Comment (emilyhcliu) [reply src/gsi/setuprad.f90:2586]: Yes!

**Comment by emilyhcliu** [comment]
Test performed :
(1) I created AMSU-A N19 and MetOp-A geoval and obs files from this feature branch (feature/jedi-gdas-radiance) for 2020110112 (global data set) 
(2) I ran the Obs Operator tests and they passed:

AMSU-A MetOp-A
```
Test     : Vector difference between reference and computed: amsua_metop-a nobs= 135645 Min=-4.65525e-05, Max=5.39063e-05, RMS=6.10721e-06
```


AMSU-A NOAA-19
```
Test     : Vector difference between reference and computed: amsua_n19 nobs= 116985 Min=-5.19284e-05, Max=4.5543e-05, RMS=6.39963e-06
```

**Comment by emilyhcliu** [comment]
@CoryMartin-NOAA 
I created the geoval and obs files for AMSU-A for testing using the latest ioda-converters.
I found that I have to include the following two lines in the setuprad.f90:
```
call nc_diag_metadata("Forecast_unadjusted", sngl(tb_obs(ich_diag(i))-tbcnob(ich_diag(i)))) ! simulated Tb with no bias correction
call nc_diag_metadata("Forecast_adjusted",   sngl(tb_obs(ich_diag(i))-tbc(ich_diag(i))))    ! simulated Tb with bias correction
```
So, the ioda-converters will use the Forecast_unadjusted as GsiHofX. 

If I did not include Foreast_unadjusted and Forecast-adjusted in the nc-diagnostic file, the ioda-converters will use "Obs_Minus_Forecast_unadjusted" and "Obs_Minus_Forecast_adjusted" to get the GsiHofX and GsiHofXBc. 
The Obs Operator did not pass:

AMSU-A NOAA-19
```
Test     : Vector difference between reference and computed: amsua_n19 nobs= 116985 Min=-254.523, Max=5.90648e-05, RMS=3.60941
```

AMSU-A MetOp-A
```
Test     : Vector difference between reference and computed: amsua_metop-a nobs= 135645 Min=-260.628, Max=5.39063e-05, RMS=80.551
```

I will look into the ioda-converters.

**Comment by CoryMartin-NOAA** [comment]
@emilyhcliu why are they different? I would think they should be the same... (O - (O-F))

**Comment by emilyhcliu** [comment]
I am testing the ozone data now!

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**